.github/workflows/ci.yaml

```diff
@@ -2,31 +2,79 @@ name: CI
 
 on: [push]
```

Comment:
tough to say, I was using it through my development in terraform module. So it might be useful when developing locally, I think we can assume for now that we will develop python code separately and then push it, and later update terraform.

Reply:
should we have separate versioning for the python code and terraform module?

Reply:
Im primarily thinking we should protect this from pushing on a public PR, either here, through its creds, or through repo configuration, but we want to prevent a random user from using the credentials.

Reply:
> should we have separate versioning for the python code and terraform module?

Possibly, but might complicate the release workflow. I think we can keep them the same for now.

Reply:
This shouldn't be how we test usually, pushing to an s3 bucket on dev should be something done locally to test, not push to test.

Reply:
Please make ci.yml to validate lint and format terraform and run test cases if there is any but not to push to s3 bucket that you use for testing. it is a public PR, noone contributing would have access to this internal s3 bucket if he wants to contribute
we shouldn't block the PR though, let's talk about testing in a follow up PR please

Reply:
sounds good, we can move on with this one, and create a new PR for the CI.